### DISCUSSION TITLE
How to allow `Vercel` deployments from `wei/pull` app?

### DISCUSSION THREAD

**Discussion by paddyroddy** [anuraghazra/github-readme-stats#3468] [Q&A]
As [per your suggestion](https://github.com/anuraghazra/github-readme-stats#keep-your-fork-up-to-date), I'm using https://github.com/wei/pull to keep my fork up-to-date - thanks for the suggestion! However, I just tried, and a PR got made, but `Vercel` couldn't deploy for permissions reasons. I read that [you need to add commit author to `Vercel`](https://vercel.com/docs/deployments/git#deploying-forks-of-public-git-repositories). Would this be the `pull[bot]`, or in my case paddyroddy/github-readme-stats#3 `qwerty541`?

**Comment by qwerty541** [comment]
Hey, @paddyroddy! Thanks for your question! At first i should say that personally i don't use `wei/pull` application, preferring to deploy vercel instance manually pushing changes from repository clone on my local machine. And as far as I know, no one from the core team uses this. I looked at the available settings for the Vercel project and became interested in the "Deploy Hooks" feature. Looks like you can deploy selected branch by visiting specially generated link after merging pull requests created by `wei/pull`. I hope that it can be solution for your problem.

![image](https://github.com/anuraghazra/github-readme-stats/assets/53787217/4fc57fd3-0f6d-4411-89e3-546b5cb9ccbc)

**Reply by paddyroddy** [reply]
Thanks, @qwerty541! `wei/pull` just operated again - and this time it got automerged. So wonder if it was a one-time approval, and then it just works.

**Reply by paddyroddy** [reply]
![image](https://github.com/anuraghazra/github-readme-stats/assets/15052188/c78fbb46-fc1c-4fdc-aa57-c249f7715492)
I found the setting, i.e. https://vercel.com/paddyroddy/github-readme-stats/settings/security

**Reply by qwerty541** [reply]
It need to be tested, but i assume that disabling git fork protection can be very unsafe in case of github-readme-stats project because of possible GitHub API token leak. If any pull request on your fork will be deployed without authorization, it means that someone can add `process.env["PAT_1"]` into svg output, open pull request to your fork and steal your GItHub API token. I highly recommend to enable this option back.

**Reply by paddyroddy** [reply]
Okay, thanks for the heads-up. Just kind of annoying that it basically means there's no benefit of `wei/pull`.

**Reply by paddyroddy** [reply]
If I disable deployments from PRs, do you think that is safe with the above setting @qwerty541?

![image](https://github.com/anuraghazra/github-readme-stats/assets/15052188/96561c89-531f-4089-abd1-0ad51b2c14d5)
![image](https://github.com/anuraghazra/github-readme-stats/assets/15052188/dd70bff7-7b1f-4cb6-b73b-8dfca7b0e7c9)

**Reply by rickstaa** [reply]
Thanks for your question, @paddyroddy! It looks like @qwerty541 has hit the nail on the head regarding the fork protection. Adding a PAT to Vercel means disabling fork protection introduces vulnerabilities. Interestingly, when I used it a while back, enabling it wasn't necessary because the project would deploy to Vercel upon merging into the master branch (see https://github.com/rickstaa/github-readme-stats/pull/86 and https://vercel.com/rickstaa/github-readme-stats/C6gedsZUFboTzpdwTzTMwcP9VMu8). I'm curious—is there any specific reason you also want it to deploy on pull requests? 🤔 I added https://github.com/wei/pull to the readme to allow people to merge upstream changes into their master manually. You can use https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request or a simple GitHub action if you want these pull requests to be merged automatically.

**Reply by paddyroddy** [reply]
I don't actually want it to deploy on pull requests! Only on merges to `main`. It seems to be the default behaviour, and not sure how to turn it off. It's now "off", but still registers on GitHub Actions as "cancelled".

**Reply by paddyroddy** [reply]
But I think the current setting means It's now safe with disabling fork protection, as deployments will only get made on `main`

**Reply by rickstaa** [reply]
@paddyroddy deploying on the main branch should work out of the box. If you don't want to deploy on the pull request, you can simply ignore the warning or follow the answer here to [disable builds on PRs](https://github.com/vercel/vercel/discussions/6434).

![image](https://github.com/anuraghazra/github-readme-stats/assets/17570430/1e216ae7-84a2-4de9-a221-2b83287578ee)

**Reply by paddyroddy** [reply]
Yep, I've already done that here https://github.com/anuraghazra/github-readme-stats/discussions/3468#discussioncomment-7773926

**Reply by qwerty541** [reply]
> If I disable deployments from PRs, do you think that is safe with the above setting @qwerty541?
>
> ![image](https://github.com/anuraghazra/github-readme-stats/assets/15052188/96561c89-531f-4089-abd1-0ad51b2c14d5)
> ![image](https://github.com/anuraghazra/github-readme-stats/assets/15052188/dd70bff7-7b1f-4cb6-b73b-8dfca7b0e7c9)

@paddyroddy When pull requests deployments are disabled, I don't see any vulnerabilities which can be caused by disabling git fork protection.

**Reply by paddyroddy** [reply]
Cheers!
